### PR TITLE
Add models and model CRUD unit tests

### PR DESCRIPTION
**Pull request by EJRyan89**
Contains basic structure of core models and unit tests for CRUD operations.

### CODE REVIEW

File: app/models/group.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Represents a group of users.
+class Group < ApplicationRecord
+  # Associations
+  has_and_belongs_to_many :users
+  has_one :list # Only have one if the group contains viewers for a list.
```
Comment (ghost) [inline app/models/group.rb:5]: What's this relationship here? Are we envisioning groups as

1. A person shares their singular list with a group of people
2. Several people join a group and share their lists with one another

I was originally envisioning our data model as 2 as that would make it easier for people to mark and view what's been purchased for others. Groups are more of an iteration 2 goal so we can discuss further.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    super()
+    @email = email
+    write_attribute(:email_hash, email) # Need hash method
+    @password = password
+    write_attribute(:password_hash, password) # Need hash method
+  end
+
+  # Override of User.create to allow for the instance to be created
+  # using content not stored in the database.
+  def self.create(email, password)
```
Comment (ghost) [inline app/models/user.rb:41]: What's this override for?

**Review by ghost — APPROVED** [review]
This looks good. I think we should talk about the Group data model a little bit, but otherwise this looks good to merge.

File: app/models/user.rb (hunk 2 above)
Comment (EJRyan89) [reply app/models/user.rb:41]: The initialize method is only called on User.new. Calling user.create requires the email and password hashes to be provided in order to create the record. The override allows for the functionality of the create method without requiring the hashes to already be generated (looking to handle that within the model, as we only really care about it for data storage).

File: app/models/group.rb (hunk 1 above)
Comment (EJRyan89) [reply app/models/group.rb:5]: The relationship here allows for the group to be associated with a given list, but does not require that all groups be associated to a list (i.e. if we wanted to be able to use a group for something other than just establishing viewers for a group, we may not require a corresponding list).

**Pull request merged**